### PR TITLE
Error bar and update to conditions 2.0

### PR DESCRIPTION
**Pull request by pmadria3**
CTW-702 and CTW-744

This PR adds the error state to the drawer and table for conditions 1.0. Additionally, conditions table 2.0 is now updated with the patient history updates. 

Error state Conditions 1.0: 
Screenshot 2023-02-07 at 2 20 49 PM

error state Conditions 2.0: 
Screenshot 2023-02-07 at 2 20 06 PM

### CODE REVIEW

**Comment by luthfib** [comment]
Isn't the error bar supposed to have a background?

File: src/components/content/conditions/patient-conditions-tabs.tsx [hunk 1]
```diff
@@ -39,9 +42,19 @@ export function PatientConditionsTabs({
   }
 
   const sharedTabStyles = "ctw-relative ctw-tab-underline ctw-tab";
+  const patientHistory = usePatientHistory();
+  const [isOther, setIsOther] = useState(false);
```
Comment (luthfib) [inline src/components/content/conditions/patient-conditions-tabs.tsx:46]: What is `isOther`?

**Comment by pmadria3** [comment]
Good catch - I renamed to text instead of background when cleaning it

File: src/components/content/conditions/patient-conditions-tabs.tsx (hunk 1 above)
Comment (pmadria3) [reply src/components/content/conditions/patient-conditions-tabs.tsx]: renamed

File: src/components/content/conditions/patient-conditions-tabs.tsx [hunk 2]
```diff
@@ -39,9 +42,19 @@ export function PatientConditionsTabs({
   }
 
   const sharedTabStyles = "ctw-relative ctw-tab-underline ctw-tab";
+  const patientHistory = usePatientHistory();
+  const [isOtherProvider, setIsOtherProvider] = useState(false);
```
Comment (luthfib) [inline src/components/content/conditions/patient-conditions-tabs.tsx:46]: Remove this we already have collection to detect this

File: src/components/content/conditions/patient-conditions-tabs.tsx [hunk 3]
```diff
@@ -39,9 +42,19 @@ export function PatientConditionsTabs({
...
   return (
     <div className="ctw-justify-end ctw-space-x-2 ctw-border-0 ctw-border-b ctw-border-t ctw-border-solid ctw-border-divider-light">
+      {isOtherProvider && (
+        <div className="ctw-space-y-3">
+          <PatientHistoryStatus
+            status={patientHistory.lastStatus}
+            date={patientHistory.dateCreatedAt}
+          />
+        </div>
+      )}
```
Comment (luthfib) [inline src/components/content/conditions/patient-conditions-tabs.tsx:57]: use collection instead

File: src/components/content/conditions/patient-conditions-tabs.tsx [hunk 4]
```diff
@@ -61,7 +77,10 @@ export function PatientConditionsTabs({
               Condition List
             </Tab>
             <Tab
-              onClick={blurClicked}
+              onClick={() => {
+                setIsOtherProvider(true);
```
Comment (luthfib) [inline src/components/content/conditions/patient-conditions-tabs.tsx:81]: Remove this new code, its unecssary

File: .changeset/good-jars-divide.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@zus-health/ctw-component-library": minor
+---
+
+Add error state to drawer and table in conditions 1.0 and conditions 2.0.
```
Comment (luthfib) [inline .changeset/good-jars-divide.md:5]: What error state? Need to be clear that is is error state for patient history

Comment (pmadria3) [inline .changeset/good-jars-divide.md:5]:
```suggestion
Add error state for patient history to drawer and table in conditions 1.0 and conditions 2.0.
```

File: src/components/content/patient-history/patient-history-table-header.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { formatISODateStringToDate } from "@/fhir/formatters";
+
+export type PatientHistoryTableHeaderMessageProps = {
+  patientHistory?: {
+    openHistoryRequestDrawer: () => Promise<void>;
+    lastRetrievedAt: string | undefined;
+    lastStatus: string | undefined;
+    dateCreatedAt: string | undefined;
```
Comment (luthfib) [inline src/components/content/patient-history/patient-history-table-header.tsx:8]: Im guessing what you wanted to do here is just make these properties optional?

Comment (pmadria3) [reply src/components/content/patient-history/patient-history-table-header.tsx]: yeah - TS did it automatically so fixed it now with ?

File: .changeset/good-jars-divide.md (hunk 5 above)
Comment (luthfib) [inline .changeset/good-jars-divide.md:5]:
```suggestion
Add error state banner to drawer component and condition 1.0 and 2.0 tables to indicate when patient history retrieval is in a failed state. 
```

**Review by luthfib — APPROVED** [review]

**Review by luthfib — APPROVED** [review]

**Pull request merged**